src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -55,6 +56,27 @@ internal ProxyFunctionExecutor(WebScriptHostManager scriptHostManager, WebHookRe
                 return await functionRequestInvoker.ProcessRequestAsync(req, ct, _scriptHostManager, _webHookReceiverManager);
             };
 
+            // Local Function calls do not go thru ARR, so implementing the ARR's MAX-FORWARDs header logic here to avoid infinte redirects.
+            IEnumerable<string> values = null;
+            int redirectCount = 0;
+            if (request.Headers.TryGetValues(ScriptConstants.AzureProxyFunctionLocalRedirectHeader, out values))
+            {
+                int.TryParse(values.FirstOrDefault(), out redirectCount);
+
+                if(redirectCount >= ScriptConstants.AzureProxyFunctionMaxLocalRedirects)
+                {
+                    response = request.CreateErrorResponse(HttpStatusCode.BadRequest, "Infinite loop detected when trying to call a local function or proxy from a proxy.");
```

Comment:
Not sure this should be a 400, since it isn't a problem with input - it's a backend issue. Shouldn't a 5xx code be more appropriate? What about 508 "Loop Detected"?

Reply:
Same behavior as IIS ARR, They return 400. And also Proxy users currently get a 400 if there is a loop in external calls.